lib/Sema/ConstraintSystem.cpp

```diff
@@ -2469,7 +2469,10 @@ bool ConstraintSystem::isPartialApplication(ConstraintLocator *locator) {
 
   auto baseTy =
       simplifyType(getType(UDE->getBase()))->getWithoutSpecifierType();
-  return getApplicationLevel(*this, baseTy, UDE) < 2;
+  auto level = getApplicationLevel(*this, baseTy, UDE);
+  // Static members have base applied implicitly which means that their
+  // application level is lower.
+  return level < (baseTy->is<MetatypeType>() ? 1 : 2);
```

Comment:
Can you explain "static members have base applied implicitly"?

Reply:
Another way to think about it is that instance methods could be referenced on metatypes - that produces a value that requires double-apply (base instance + parameters) to be fully applied, but statics can only be accessed on metatypes and because of that the base is always applied.

Reply:
Ok I see, so static members base is not included in the "number of applications", thanks!